src/sentry/sentry_apps/api/serializers/sentry_app_component.py

```diff
@@ -14,7 +14,7 @@ def serialize(self, obj, attrs, user, **kwargs):
             "uuid": str(obj.uuid),
             "type": obj.type,
             "schema": obj.schema,
-            "error": True if str(obj.uuid) in errors else False,
+            "error": errors.get(str(obj.uuid), ""),
```

Comment:
In the [above](https://github.com/getsentry/sentry/pull/81167/files#diff-00e9151c57a98b8e0d198312c2dcdb6eca386cc3936f121345e427deb738b8a7R85-R95) we catch all errors (assuming they're raised), so there should never be the case where a component in in `errors` but the message is an empty string.

Reply:
what does it look like on the frontend when we return the entire error instead of a bool?

Reply:
Nothing will change for frontend appearance wise. Currently we use that boolean to determine whether or not we [need to gray out the UI component](https://github.com/getsentry/sentry/blob/master/static/app/components/sentryAppComponentIcon.tsx#L39-L41) and display a generic 'couldn't connect to [SentryApp]' message. That message and appearance will stay the same, the new errors will be exposed via the network requests.

Reply:
should that be a change we make -- to surface the more explicit error message?

Reply:
we could, but that would be a future PR. Our integrators are all developers who will check the network reqs to debug cust issues, so this change is needed either way to provide that visibility.